### PR TITLE
Add network aliases for containers to DB

### PR DESCRIPTION
**Pull request by mheon**
This adds the database backend for network aliases. Aliases are additional names for a container that are used with the CNI dnsname plugin - the container will be accessible by these names in addition to its name. Aliases are allowed to change over time as the container connects to and disconnects from networks.

Aliases are implemented as another bucket in the database to register all aliases, plus two buckets for each container (one to hold connected CNI networks, a second to hold its aliases). The aliases are only unique per-network, to the global and per-container aliases buckets have a sub-bucket for each CNI network that has aliases, and the aliases are stored within that sub-bucket. Aliases are formatted as alias (key) to container ID (value) in both cases.

Three DB functions are defined for aliases: retrieving current aliases for a given network, setting aliases for a given network, and removing all aliases for a given network.

This is still only partially done. Remaining work:
- Implement the remove aliases function for BoltDB
- Implement aliases in the in-memory state
- Add unit testing for the new aliases functionality in the DB
- Add accessors to Container to get aliases
- Add an accessor to get the container's IP for a specific CNI network

### CODE REVIEW

File: libpod/boltdb_state_internal.go [hunk 1]
```diff
@@ -46,26 +48,28 @@ const (
...
-	podIDKey           = []byte(podIDName)
-	namespaceKey       = []byte(namespaceName)
+	networksBkt        = []byte(networksName)
+
+	configKey     = []byte(configName)
+	stateKey      = []byte(stateName)
+	netNSKey      = []byte(netNSName)
+	containersBkt = []byte(containersName)
+	podIDKey      = []byte(podIDName)
+	namespaceKey  = []byte(namespaceName)
```
Comment (TomSweeneyRedHat) [inline libpod/boltdb_state_internal.go:72]: any chance these could be alphabetized?

File: libpod/boltdb_state.go [hunk 2]
```diff
@@ -969,6 +972,184 @@ func (s *BoltState) AllContainers() ([]*Container, error) {
...
+			return nil
+		})
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return aliases, nil
+}
+
```
Comment (TomSweeneyRedHat) [inline libpod/boltdb_state.go:1037]: Maybe I missed it, but will there be a `podman list network aliases {network}` or similar command/option?

Comment (mheon) [reply libpod/boltdb_state.go:1037]: No. They'll show up in `podman inspect` on the container they're attached to.

**Comment by mheon** [comment]
Stripping WIP. Still missing some bits, but we can add them as part of the future Network Connect / Network Disconnect work.

**Comment by rhatdan** [comment]
@baude PTAL

**Comment by baude** [comment]
@rhatdan PTAL

File: libpod/boltdb_state_internal.go (hunk 1 above)
Comment (mheon) [reply libpod/boltdb_state_internal.go:72]: Right now they're in the order they're created in, DB-wise.

**Comment by baude** [comment]
LGTM and i verified this with the aliases code and it all still works.

**Comment by ashley-cui** [comment]
LGTM (from an initial read-through)

**Comment by mheon** [comment]
This is green
@baude

**Pull request merged**